src/components/accordion-disclosure/index.tsx

```diff
@@ -11,18 +12,48 @@ import {
 	useAccordionDisclosureContext,
 	AccordionDisclosureContext,
 } from './accordion-disclosure-context'
+import useHover from 'hooks/use-hover'
 
 const AccordionDisclosure = ({
 	children,
 	description,
 	initialOpen,
 	title,
+	groupData,
+	activatorHeadingLevel,
 }: AccordionDisclosureProps) => {
+	/**
+	 * Note: useHover is used here to allow us to change the styling
+	 * of the outer `root` container component when the `<button />`
+	 * within, which is the functional interactive element, is hovered.
+	 *
+	 * Since we're changing styles on an ancestor node of the `<button />`
+	 * whose `:hover` state we want to track, this is not as feasible
+	 * with pure CSS.
+	 */
+	const [hoverRef, isHovered] = useHover<HTMLButtonElement>()
 	const isNested = useAccordionDisclosureContext()
+
+	/**
+	 * Set up variables related to group styling.
+	 */
+	const isGroupedItem = typeof groupData !== 'undefined'
+	const isFirstItem = groupData?.currentIndex === 0
+	const isLastItem = groupData?.currentIndex === groupData?.numItems - 1
```

Comment:
I think we should use CSS selectors for the first/last item styles, since this purely has to do with styles instead of functionality. The JS approach also requires that the correct data is passed in for each of these components, whereas the CSS approach would automatically handle styles based on placement and be less work on the DX side.

Reply:
I like this suggestion! I tried working through it, but hit a few dead ends.

It doesn't seem feasible to me to write styles _within_ `AccordionDisclosure` for first & last styles. For styling the first element in a group, for example, as far as I know we can only target `:first-child` or `:first-of-type`, and the root element here is type `div`. I think we'd have many scenarios where, if we write styles using these selectors, we'd get false positives and see group-like styles in contexts where `AccordionDisclosure` is next to another `div`, for example. (We might be able to use sibling selectors for the middle elements reliably... but the first and last elements seem like a challenge!).

Another option might be reaching into the component and styling it from the outside. We could potentially add styles to the `Accordion` component that target elements within `AccordionDisclosure`, and in that case `:first-child`, `:nth-child`, `:last-child` etc seem like they would work. But, we'd either have to use relatively brittle global-ish selectors (eg `.accordionRoot > div:first-child` or whatnot), or replace the `groupData` prop with a `styleClassName` prop or something that we'd tack onto the element rendering the box shadow.

Personally I think the `groupData` prop approach is slightly preferable to styling from the outside:
- We have a clear and limited set of styles for `AccordionDisclosure`, and the `groupData` prop provides clear documentation as to why these styles exist. Styles that affect the component elements are all contained within the component itself, rather than spread in multiple places in the app.
- When iterating on the component, we don't have to audit and account for all use cases that styling from the outside might be used for. For example, when passing a `styleClassName` prop, as we iterate on the `AccordionDisclosure` styles we might inadvertently change styles in a way that breaks or negatively affects the `styleClassName` styles being passed in by a consumer.

Let me know if the `groupData` prop here seems like an acceptable solution! If we feel like it's blocking, and we want to explore other approaches, I'd be very open to suggestions 🙇